sdk/src/service.rs

```diff
@@ -291,3 +302,145 @@ where
         self.router.call(request)
     }
 }
+
+pub struct Config<Plugin> {
+    // Built final plugin
+    pub(crate) plugin: Plugin,
+}
+
+impl Config<()> {
+    pub fn builder() -> config::Builder<IdentityPlugin> {
+        config::Builder {
+            plugin_pipeline: aws_smithy_http_server::plugin::PluginPipeline::default(),
+            aws_auth_authenticate_inserted_at: None,
+            aws_auth_authorize_inserted_at: None,
+            plugin_count: 0,
+        }
+    }
+}
+
+/// Module hosting the builder for `Config`.
+pub mod config {
+    use aws_smithy_http_server::plugin::{IdentityPlugin, PluginPipeline, PluginStack};
+
+    // This follows the same pattern as `MissingOperationsError`.
+    #[derive(Debug)]
+    pub struct Error {
+        msg: String,
+    }
+
+    impl std::fmt::Display for Error {
+        fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+            writeln!(f, "{}", self.msg)?;
+            Ok(())
+        }
+    }
+
+    impl std::error::Error for Error {}
+
+    pub struct Builder<Plugin> {
+        pub(super) plugin_pipeline: PluginPipeline<Plugin>,
+
+        // There's one `usize` per thing that must be configured, tracking
+        // the index of when the plugin was inserted.
+        pub(super) aws_auth_authenticate_inserted_at: Option<usize>,
+        pub(super) aws_auth_authorize_inserted_at: Option<usize>,
+
+        pub(super) plugin_count: usize,
+    }
+
+    impl<Plugin> Builder<Plugin> {
+        /// Apply a new plugin after the ones that have already been registered.
+        pub fn plugin<NewPlugin>(
```

Comment:
With this method it seems like there'd be no need for `PluginPipeline`, since the `Config` builder has the same interface, but I think we should keep `PluginPipeline` so that third-parties can still bundle plugins together and vend them without being in the context of a code generated server SDK.